source/dub/compilers/buildsettings.d

```diff
@@ -148,7 +150,10 @@ private:
 	// Prepend arr by vals without adding duplicates.
 	static void prepend(ref string[] arr, in string[] vals, bool noDuplicates = true)
 	{
-		arr = filterDuplicates(arr, vals, noDuplicates) ~ arr;
+		import std.algorithm.mutation : reverse;
+		// vals might contain duplicates, add each val individually
+		foreach (val; vals.dup.reverse)
```

Comment:
Would the following work:

```suggestion
		import std.range : retro;
		// vals might contain duplicates, add each val individually
		foreach (val; vals.retro)
```

If it it does, it should avoid the memory allocation caused by `.dup`.

Reply:
Yep, it works. Thanks!